webapi/src/main/scala/org/knora/webapi/responders/admin/PermissionsResponder.scala

```diff
@@ -534,151 +550,66 @@ final case class PermissionsResponder(
     } yield DefaultObjectAccessPermissionsStringResponseADM(resultStr)
   }
 
-  private def validate(req: CreateDefaultObjectAccessPermissionAPIRequestADM) = ZIO.attempt {
-    val sf: StringFormatter = StringFormatter.getInstanceForConstantOntologies
+  private def validate(
+    req: CreateDefaultObjectAccessPermissionAPIRequestADM,
+  ): IO[String, DefaultObjectAccessPermission] =
+    for {
+      projectIri <- ZIO.fromEither(ProjectIri.from(req.forProject))
+      project    <- knoraProjectService.findById(projectIri).orDie.someOrFail("Project not found")
+      permissionIri <-
+        ZIO
+          .foreach(req.id)(iriConverter.asSmartIri)
+          .flatMap(iriService.checkOrCreateEntityIri(_, PermissionIri.makeNew(project.shortcode).value))
+          .mapBoth(_.getMessage, PermissionIri.unsafeFrom)
 
-    req.id.foreach(iri => PermissionIri.from(iri).fold(msg => throw BadRequestException(msg), _ => ()))
+      groupIri <- ZIO.fromEither(req.forGroup.fold(Right(None))(GroupIri.from(_).map(Some(_))))
+      _        <- ZIO.foreachDiscard(groupIri)(groupService.findById(_).orDie.someOrFail("Group not found"))
 
-    ProjectIri.from(req.forProject).getOrElse(throw BadRequestException(s"Invalid project IRI  ${req.forProject}"))
+      resourceClassIri <-
+        ZIO.foreach(req.forResourceClass)(
+          iriConverter.asSmartIri(_).mapError(_.getMessage).flatMap(s => ZIO.fromEither(ResourceClassIri.from(s))),
+        )
 
-    (req.forGroup, req.forResourceClass, req.forProperty) match {
-      case (None, None, None) =>
-        throw BadRequestException(
-          "Either a group, a resource class, a property, or a combination of resource class and property must be given.",
+      propertyIri <-
+        ZIO.foreach(req.forProperty)(
+          iriConverter.asSmartIri(_).mapError(_.getMessage).flatMap(s => ZIO.fromEither(PropertyIri.from(s))),
```

Comment:
These two foreach above allow the property and the resourceClass to be either an internal or a ApiV2Complex iri.